java/adapter/jdbc/src/main/java/org/apache/arrow/adapter/jdbc/JdbcToArrowUtils.java

```diff
@@ -0,0 +1,343 @@
+                    fields.add(new Field(columnName, FieldType.nullable(new ArrowType.Bool()), null));
+                    break;
+                case Types.TINYINT:
+                    fields.add(new Field(columnName, FieldType.nullable(new ArrowType.Int(8, true)), null));
+                    break;
+                case Types.SMALLINT:
+                    fields.add(new Field(columnName, FieldType.nullable(new ArrowType.Int(16, true)), null));
+                    break;
+                case Types.INTEGER:
+                    fields.add(new Field(columnName, FieldType.nullable(new ArrowType.Int(32, true)), null));
+                    break;
+                case Types.BIGINT:
+                    fields.add(new Field(columnName, FieldType.nullable(new ArrowType.Int(64, true)), null));
+                    break;
+                case Types.NUMERIC:
+                case Types.DECIMAL:
+                    int precision = rsmd.getPrecision(i);
+                    int scale = rsmd.getScale(i);
+                    fields.add(new Field(columnName, FieldType.nullable(new ArrowType.Decimal(precision, scale)), null));
+                    break;
+                case Types.REAL:
+                case Types.FLOAT:
+                    fields.add(new Field(columnName, FieldType.nullable(new ArrowType.FloatingPoint(SINGLE)), null));
+                    break;
+                case Types.DOUBLE:
+                    fields.add(new Field(columnName, FieldType.nullable(new ArrowType.FloatingPoint(DOUBLE)), null));
+                    break;
+                case Types.CHAR:
+                case Types.NCHAR:
+                case Types.VARCHAR:
+                case Types.NVARCHAR:
+                case Types.LONGVARCHAR:
+                case Types.LONGNVARCHAR:
+                    fields.add(new Field(columnName, FieldType.nullable(new ArrowType.Utf8()), null));
+                    break;
+                case Types.DATE:
+                    fields.add(new Field(columnName, FieldType.nullable(new ArrowType.Date(DateUnit.MILLISECOND)), null));
+                    break;
+                case Types.TIME:
+                    fields.add(new Field(columnName, FieldType.nullable(new ArrowType.Time(TimeUnit.MILLISECOND, 32)), null));
+                    break;
+                case Types.TIMESTAMP:
+                    // timezone is null
+                    fields.add(new Field(columnName, FieldType.nullable(new ArrowType.Timestamp(TimeUnit.MILLISECOND, null)), null));
+                    break;
+                case Types.BINARY:
+                case Types.VARBINARY:
+                case Types.LONGVARBINARY:
+                    fields.add(new Field(columnName, FieldType.nullable(new ArrowType.Binary()), null));
+                    break;
+                case Types.ARRAY:
+                    // not handled
+//                    fields.add(new Field("list", FieldType.nullable(new ArrowType.List()), null));
```

Comment:
field is ignored, so doesn't that mess up with the resulting schema? (number of fields different between JDBC and Arrow)?

Reply:
In this case, Array type is ignored for the schema creation as well as Data creation. Let me spend some time on how I can model the Array data inside Arrow Vectors? Any clues?